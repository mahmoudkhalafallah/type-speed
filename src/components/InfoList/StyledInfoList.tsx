import React from 'react'
import styled from 'styled-components'
import Time from './Time'
import WPM from './WPM'
import CP from './CP'
import RecordLink from './RecordLink'

const StyledList = styled.ul`
align-self: flex-end;
text-align: end;
margin: 25px 0;
list-style: none;
font-family: "capture_it";
`
const StyledItem = styled.li`
margin-bottom: 5px;
span {
    min-width: 70px;
    display: inline-block;
}
a {
    color: #3d3d3d;
}
`

interface Props {
    time: { minutes: number, seconds: number };
    gameOver: boolean;
    wpm: number;
    cp: number;
    historyId: string
}

const StyledInfoList: React.FC<Props> = ({ time, gameOver, wpm, cp, historyId }) => {
    return <StyledList>
        <StyledItem>
            <Time minutes={time.minutes} seconds={time.seconds} />
        </StyledItem>
        <StyledItem>
            <WPM wpm={wpm} />
        </StyledItem>
        {gameOver && <StyledItem>
            <CP cp={cp} />
        </StyledItem>}
        {(historyId.length > 0) && <StyledItem>
            <RecordLink historyId={historyId} />
        </StyledItem>}
    </StyledList>
}

export default StyledInfoList