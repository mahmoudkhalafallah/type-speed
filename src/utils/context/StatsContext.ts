import React from 'react'
const StatsContext = React.createContext({ wpm: 0, cp: 0, historyId: '' })
export default StatsContext