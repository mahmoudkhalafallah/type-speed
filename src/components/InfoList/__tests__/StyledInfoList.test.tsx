import React from 'react'
import StyledInfoList from '../StyledInfoList'
import { render } from '@testing-library/react'
import StatsContext from '../../../utils/context/StatsContext'


test('StyledInfoList rendered correctly', () => {
    const { baseElement } = render(
        <StatsContext.Provider value={{ wpm: 0, cp: 0, historyId: '' }}>
            <StyledInfoList
                showCp={false}
                time={{ minutes: 3, seconds: 0 }}
            />
        </StatsContext.Provider>
    )
    expect(baseElement).toMatchSnapshot()
})

test('StyledInfoList rendered correctly after time is up', () => {
    const { baseElement } = render(
        <StatsContext.Provider value={{ wpm: 30, cp: 80, historyId: 'xyz' }}>
            <StyledInfoList
                showCp={true}
                time={{ minutes: 0, seconds: 0 }}
            />
        </StatsContext.Provider>,
    )
    expect(baseElement).toMatchSnapshot()
})