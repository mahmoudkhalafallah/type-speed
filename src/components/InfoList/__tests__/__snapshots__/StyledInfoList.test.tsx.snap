// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInfoList rendered correctly 1`] = `
<body>
  <div>
    <ul
      class="sc-bwzfXH kkRSrs"
    >
      <li
        class="sc-htpNat hCuCUW"
      >
        Time:
        <span>
          3
          :
          00
        </span>
      </li>
      <li
        class="sc-htpNat hCuCUW"
      >
        Words Per Minute:
        <span
          class="sc-bdVaJa eBZPoX"
        >
          0
           WPM
        </span>
      </li>
    </ul>
  </div>
</body>
`;

exports[`StyledInfoList rendered correctly after time is up 1`] = `
<body>
  <div>
    <ul
      class="sc-bwzfXH kkRSrs"
    >
      <li
        class="sc-htpNat hCuCUW"
      >
        Time:
        <span>
          0
          :
          00
        </span>
      </li>
      <li
        class="sc-htpNat hCuCUW"
      >
        Words Per Minute:
        <span
          class="sc-bdVaJa eDGlOc"
        >
          30
           WPM
        </span>
      </li>
      <li
        class="sc-htpNat hCuCUW"
      >
        Completion Percentage:
        <span>
          80
          %
        </span>
      </li>
      <li
        class="sc-htpNat hCuCUW"
      >
        Find your record 
        <a
          data-testid="recordLink"
          href="/history/xyz"
        >
          here
        </a>
      </li>
    </ul>
  </div>
</body>
`;
